### PR TITLE
Enabled type-spec/go emitter

### PR DESCRIPTION
**Pull request by raych1**
Enabled go emitter for all the valid services.

### CODE REVIEW

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml [hunk 1]
```diff
@@ -37,3 +37,13 @@ options:
...
+  "@azure-tools/typespec-go":
+    service-dir: "sdk/resourcemanager/computefleet"
+    package-dir: "armcomputefleet"
+    module: github.com/Azure/azure-sdk-for-go/{service-dir}/{package-dir}
+    flavor: "azure"
+    generate-fakes: true
+    head-as-boolean: true
+    inject-spans: true
+    remove-unreferenced-types: true
+    examples-directory: ./examples
```
Comment (tadelesh) [inline specification/azurefleet/AzureFleet.Management/tspconfig.yaml:49]:
```suggestion
    examples-directory: "{project-root}/examples"
```

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml [hunk 2]
```diff
@@ -37,3 +37,13 @@ options:
     service-name: Compute Fleet
     examples-directory: "examples"
     flavor: azure
+  "@azure-tools/typespec-go":
+    service-dir: "sdk/resourcemanager/computefleet"
+    package-dir: "armcomputefleet"
+    module: github.com/Azure/azure-sdk-for-go/{service-dir}/{package-dir}
```
Comment (tadelesh) [inline specification/azurefleet/AzureFleet.Management/tspconfig.yaml:43]:
```suggestion
    module: "github.com/Azure/azure-sdk-for-go/{service-dir}/{package-dir}"
```

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml (hunk 1 above)
Comment (tadelesh) [inline specification/azurefleet/AzureFleet.Management/tspconfig.yaml:49]: add several configs: 
generate-examples: true
fix-const-stuttering: true

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml [hunk 3]
```diff
@@ -37,3 +37,13 @@ options:
...
+    flavor: "azure"
```
Comment (tadelesh) [inline specification/azurefleet/AzureFleet.Management/tspconfig.yaml:44]:
```suggestion
```

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml (hunk 1 above)
Comment (raych1) [reply specification/azurefleet/AzureFleet.Management/tspconfig.yaml:49]: @tadelesh , is `generate-examples` a new option of go emitter? I don't find it in the code.

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml (hunk 3 above)
Comment (raych1) [reply specification/azurefleet/AzureFleet.Management/tspconfig.yaml:46]: We will keep this as TypeSpec validation tool validate this requirement.

File: specification/azurefleet/AzureFleet.Management/tspconfig.yaml (hunk 1 above)
Comment (raych1) [reply specification/azurefleet/AzureFleet.Management/tspconfig.yaml]: Added them as release tool won't do schema validation for emitter options. This option will be added later to emitter code.

**Review by tadelesh — APPROVED** [review]

**Pull request merged**